src/adapter/debugAdapter.ts

```diff
@@ -173,7 +173,37 @@ export class DebugAdapter implements IDisposable {
     this._pauseOnExceptionsState = 'none';
     if (params.filters.includes('caught')) this._pauseOnExceptionsState = 'all';
     else if (params.filters.includes('uncaught')) this._pauseOnExceptionsState = 'uncaught';
-    if (this._thread) await this._thread.setPauseOnExceptionsState(this._pauseOnExceptionsState);
+    if (this._thread) {
+      const result = this._thread.setPauseOnExceptionsState(this._pauseOnExceptionsState);
+      const logger = this._services.get<ILogger>(ILogger);
+      if (this._configurationDoneDeferred.hasSettled()) {
+        logger.verbose(LogTag.Internal, 'setExceptionBreakpoints: awaiting blocking response');
+        await result;
+      } else {
+        result
+          .then(() => {
+            logger.verbose(
+              LogTag.Internal,
+              'setExceptionBreakpoints: succesful non-blocking response',
+            );
+          })
+          .catch(rejection => {
+            const output = localize(
+              '`errors.setExceptionBreakpoints.async.failed`',
+              'Failed to configure the exceptions for which to pause due to: {0}',
+              rejection.message || localize('errors.unknown', 'Unknown error'),
+            );
+            this.dap.output({ category: 'stderr', output: output });
+            const telemetry = this._services.get<ITelemetryReporter>(ITelemetryReporter);
+            telemetry.reportOperation(
```

Comment:
This operation data is already reported by the CDP connection itself, I don't think we need extra bits here

Reply:
You are right